src/eva/data/ioda_obs_space.py

```diff
@@ -64,23 +71,53 @@ def execute(self, data_collections):
                 # Loop over filenames
                 for filename in filenames:
 
-                    # Open file and add collection
-                    with xr.open_dataset(filename, group=group) as ds:
+                    # Read header part of the file to get coordinates
```

Comment:
Rahul will say to do a with open like you had before :-)

Reply:
Personally, I think it's fine the way it is. But I know that the 'pythonic' way is the 'with ...'. But leave it as is, readability is important.

Reply:
OK, will do. We can look again at it. Especially if the code changes as we learn more about the way it works when I start adding filters and trying to make plots.